### PR TITLE
Add mint to YASPool

### PR DESCRIPTION
**Pull request by dpinones**
Closes: #89

### CODE REVIEW

File: src/contracts/yas_pool.cairo [hunk 1]
```diff
@@ -429,10 +461,91 @@ mod YASPool {
...
+            recipient: ContractAddress,
+            tick_lower: i32,
+            tick_upper: i32,
+            amount: u128,
+            data: Array<felt252>
+        ) -> (u256, u256) {
+            self.check_and_lock();
+
+            assert(amount > 0, 'amount must be greater than 0');
+            let (_, amount_0, amount_1) = self
```
Comment (rcatalan98) [inline src/contracts/yas_pool.cairo:476]: If we are not using the first element returned, why are returning in it? This function is only used in `mint` right?

File: src/contracts/yas_pool.cairo [hunk 2]
```diff
@@ -429,10 +461,91 @@ mod YASPool {
...
+                    }
+                );
+
+            let amount_0: u256 = amount_0.try_into().unwrap();
+            let amount_1: u256 = amount_1.try_into().unwrap();
+
+            let mut balance_0_before = 0;
+            if amount_0 > 0 {
+                balance_0_before = self.balance_0();
+            }
```
Comment (rcatalan98) [inline src/contracts/yas_pool.cairo:490]:
```suggestion
           let balance_0_before = if amount_0 > 0 {
    self.balance_0()
};
```

File: src/contracts/yas_pool.cairo [hunk 3]
```diff
@@ -429,10 +461,91 @@ mod YASPool {
...
+
+            let mut balance_1_before = 0;
+            if amount_1 > 0 {
+                balance_1_before = self.balance_1();
+            }
```
Comment (rcatalan98) [inline src/contracts/yas_pool.cairo:495]:
```suggestion
           let balance_1_before = if amount_1 > 0 {
    self.balance_1()
};
```

Comment (dpinones) [reply src/contracts/yas_pool.cairo]: ✅

File: src/contracts/yas_pool.cairo (hunk 2 above)
Comment (dpinones) [reply src/contracts/yas_pool.cairo]: ✅

File: src/contracts/yas_pool.cairo (hunk 1 above)
Comment (dpinones) [reply src/contracts/yas_pool.cairo:477]: The `modify_position` function is used in the `burn` function and uses the first field

**Review by rcatalan98 — APPROVED** [review]
LGTM!

**Review by dubzn — APPROVED** [review]
LGTM :cowboy_hat_face:

**Pull request merged**